### PR TITLE
Fixes to allow the plugin to work in version 3.3 and 3.4

### PR DESCRIPTION
**Pull request by andresmrg**
Changes in .travis.yml and classes doesn't have a huge impact but I consider them improvements. I improved the strings in the lang folder to make it more clear to the end user. The fixes and improvements are in the yui/ folder. Feel free to adjust it. Here is a small video of my testing https://screencast.com/t/uCOb2kVbhlZ using Moodle 3.3

### CODE REVIEW

File: classes/frontend.php [hunk 1]
```diff
@@ -46,12 +46,7 @@ class frontend extends \core_availability\frontend {
      * @param \section_info $section Section currently being edited (null if none)
      */
     protected function allow_add($course, \cm_info $cm = null, \section_info $section = null) {
-        global $add;
```
Comment (danielneis) [inline classes/frontend.php]: This part is a workaround to avoid this module to be added when using book, label and wiki, where it does not.
Have you tested this to see if it was not working anymore?

File: composer.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "name": "danielneis/moodle-availability_maxviews",
     "description": "An availability condition for Moodle that limits the number of views of an activity by users",
     "type": "moodle-availability",
-    "license": "GPL-3.0-or-later",
+    "license": "GPLv3",
     "authors": [
```
Comment (danielneis) [inline composer.json]: your commit was made on an previous version of the plugin. this license has to be this to be valid on packagist.
please, undo this change.

File: .travis.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ matrix:
 
 env:
  global:
-  - MOODLE_BRANCH=MOODLE_31_STABLE
+  - MOODLE_BRANCH=MOODLE_30_STABLE
  matrix:
```
Comment (danielneis) [inline .travis.yml]: it must be changed to MOODLE_34_STABLE instead of 30

File: classes/condition.php [hunk 4]
```diff
@@ -69,8 +63,8 @@ public function save() {
      * @param int $viewslimit The limit of views for users
      * @return stdClass Object representing condition
      */
-    public static function get_json($viewslimit = 10) {
-        return (object)array('type' => 'maxviews', 'viewslimit' => $viewslimit);
+    public static function get_json($viewslimit = 5) {
```
Comment (danielneis) [inline classes/condition.php]: why change the limit to 5 ? (although i had no reasons to set it to 10 also)

Comment (andresmrg) [reply classes/condition.php]: In the improvements that I made, this is no longer relevant or used. When adding the condition, it has an empty field but can't be left empty, zero or negative number. That's a validation I added.

File: .travis.yml (hunk 3 above)
Comment (danielneis) [reply .travis.yml]: I just made a commit to update this, if you rebase you'll be fine =)

File: classes/frontend.php (hunk 1 above)
Comment (andresmrg) [reply classes/frontend.php]: Nope, I rollback the changes here with your validation for the 3 activity modules.

File: composer.json (hunk 2 above)
Comment (andresmrg) [reply composer.json]: Done

**Comment by andresmrg** [comment]
Please review again, I made two more commits. Not quite sure how to rebase from the old one so I took what you had in travis and added it to the commit.
